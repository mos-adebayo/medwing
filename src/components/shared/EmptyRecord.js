import React from 'react';

const EmptyRecord = ({message}) => {
    return (
        <div className={'section empty-container'}>
            <h1 className={'center-align'}>
                <i className="large material-icons">info</i>
            </h1>
            <p className={'center-align'}>
                {message}
            </p>
        </div>
    );
};

export default EmptyRecord;
