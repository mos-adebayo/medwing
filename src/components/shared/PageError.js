import React from 'react';

export const ComponentErrorMessage = ({message}) => {
    return (
        <div className={'section empty-container'}>
            <h1 className={'center-align'}>
                <i className="large material-icons red-text">error</i>
            </h1>
            <p className={'center-align'}>
                {message}
            </p>
        </div>
    );
};
export const AppErrorMessage = ({message}) => {
    return (
        <div className={'app-error red'}>
            <p className={'white-text'}>
                {message}
            </p>
        </div>
    );
};
