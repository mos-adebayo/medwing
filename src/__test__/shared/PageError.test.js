import React from "react";
import { configure, shallow } from "enzyme";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import { ComponentErrorMessage, AppErrorMessage } from "../../components/shared/PageError";


//Enzyme Configurations
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const errorMessage = 'Error message';

test('AppErrorMessage components snapshot', () => {
    const component = renderer.create(
        <AppErrorMessage message={errorMessage} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('AppErrorMessage component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AppErrorMessage message={errorMessage} />, div);
    ReactDOM.unmountComponentAtNode(div);
});



test("AppErrorMessage component receives props and then renders text", () => {
    // Renders AppErrorMessage component with some text prop.
    const AppErrorMessageRender = shallow(<AppErrorMessage message={errorMessage} />);
    expect(AppErrorMessageRender.text()).toEqual(errorMessage)
});

test('ComponentErrorMessage components snapshot', () => {
    const component = renderer.create(
        <ComponentErrorMessage message={errorMessage} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('ComponentErrorMessage component renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ComponentErrorMessage message={errorMessage} />, div);
    ReactDOM.unmountComponentAtNode(div);
});



test("ComponentErrorMessage component receives props and then renders text", () => {
    const ComponentErrorMessageRender = shallow(<ComponentErrorMessage message={errorMessage} />);
    expect(ComponentErrorMessageRender.contains((
        <h1 className={'center-align'}>
            <i className="large material-icons red-text">error</i>
        </h1>
    ))).toBeTruthy();
    expect(ComponentErrorMessageRender.contains((<p className={'center-align'}>{errorMessage}</p>))).toBeTruthy()
});
