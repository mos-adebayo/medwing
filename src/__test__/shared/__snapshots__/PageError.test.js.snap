// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppErrorMessage components snapshot 1`] = `
<div
  className="app-error red"
>
  <p
    className="white-text"
  >
    Error message
  </p>
</div>
`;

exports[`ComponentErrorMessage components snapshot 1`] = `
<div
  className="section empty-container"
>
  <h1
    className="center-align"
  >
    <i
      className="large material-icons red-text"
    >
      error
    </i>
  </h1>
  <p
    className="center-align"
  >
    Error message
  </p>
</div>
`;
